Automattic/wp-calypso
# Login: Add native login into calypso

This PR adds a simple login component, connected to the redux store.
Props to Luna for this work: @johnHackworth @oskosk and @tyxla. Original PR https://github.com/Automattic/wp-calypso/pull/11367

I have excluded 2FA from the original PR for now to make the review process faster.
Also I moved the endpoint from `/jetpack-connect/login` to `/login`. This route is already used for OAuth (on the desktop app) so both `oauth` and `login-native` cannot be enabled at the same time.
Moreover the css class `is-section-${ sectionName }` is used to customize styles for a section so I had to disable the blue background for now for OAuth. There is probably a simple way to reintroduce it if needed but I have found it and I rather have a common theme for all login pages.

This is how it looks like:
![image](https://cloud.githubusercontent.com/assets/230230/24460165/bed4e582-149d-11e7-9b14-5a151aae60ea.png)

This is a first iteration which contains everything needed regarding the UI for a basic login, it does not however make login magically work, we will need to connect it to our new endpoints. You can apply patch D4910 to have an idea of how it will work but this should not be a blocker for the review.

### Testing Instructions
- Run this branch locally
- Go to `/login`
- Check that the page loads
- Set `features.oauth` to `true` and `features.login-native` to false in `config.development.json` and check that the login page for oauth still displays correctly (but is not blue anymore).
- Optional: apply patch D4910 and test that entering the right credentials work

### Reviews
